src/mono/System.Private.CoreLib/src/System/Enum.Mono.cs

```diff
@@ -40,17 +41,42 @@ internal static RuntimeType InternalGetUnderlyingType(RuntimeType enumType)
             return res!;
         }
 
-        private static EnumInfo GetEnumInfo(RuntimeType enumType, bool getNames = true)
+        /// <summary>Creates a new TUnderlyingValue[] from a ulong[] array of values.</summary>
+        private static TUnderlyingValue[] ToUnderlyingValues<TUnderlyingValue>(ulong[] uint64Values)
+            where TUnderlyingValue : struct, INumber<TUnderlyingValue>
         {
-            EnumInfo? entry = enumType.Cache.EnumInfo;
+            TUnderlyingValue[] values;
+
+            if (typeof(TUnderlyingValue) == typeof(ulong))
+            {
+                values = (TUnderlyingValue[])(object)uint64Values;
+            }
+            else
+            {
+                values = new TUnderlyingValue[uint64Values.Length];
+
+                for (int i = 0; i < values.Length; i++)
+                {
+                    values[i] = TUnderlyingValue.CreateTruncating(uint64Values[i]);
```

Comment:
This is not correct for float/double. This is converting the integer to float/double value. We need to get float/double bytes instead.